frontend/lib/Models/group_endpoint_provider.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:adpv_frontend/DataModels/group_endpoints.dart';
+import 'package:flutter/material.dart';
+
+import '../Repository/AdminRepository/groups_repository.dart';
+
+class GroupEndpointProvider with ChangeNotifier {
+  GroupEndpointsData groupEndpointsData = GroupEndpointsData.empty();
+  GroupsRepository repository = GroupsRepository();
+
+  GroupEndpointProvider(Future<GroupEndpointsData> future) {
+    init(future);
+  }
+
+  void init(Future<GroupEndpointsData> future) {
+    future.then((value) {
+      groupEndpointsData = value;
+      groupEndpointsData.sortFields();
+      notifyListeners();
+    });
+  }
```

Comment:
Czy do `Provider'a` powinniśmy przekazywać `Future`? W wcześniejszych podejściach w tym miejscu wrzucaliśmy po prostu model i ja bym tak samo robił tutaj

Reply:
tutaj wydaje mi sie uzasadnione - robiąc w ten sposób pobierzemy listę tylko raz, lista "odświeża się" z każdym jej dotknięciem

Reply:
Tylko w tym momencie rozjeżdża się nam trochę podejście, bo w innych `Provider'ach` tak nie robimy. Byłbym za tym, żeby to ujednolicić